crates/matrix-sdk-ui/src/notification_client.rs

```diff
@@ -684,30 +684,23 @@ impl NotificationClient {
                 continue;
             }
 
-            let status =
+            let notification_item =
                 match NotificationItem::new(&room, raw_event, push_actions.as_deref(), Vec::new())
                     .await
-                    .map(|event| NotificationStatus::Event(Box::new(event)))
                 {
-                    Ok(status) => status,
+                    Ok(item) => item,
                     Err(err) => {
                         // Could not build the notification item, return an error.
                         batch_result.insert(event_id, Err(err));
                         continue;
                     }
                 };
 
-            match status {
-                NotificationStatus::Event(event) => {
-                    if self.client.is_user_ignored(event.event.sender()).await {
-                        batch_result.insert(event_id, Ok(NotificationStatus::EventFilteredOut));
-                    } else {
-                        batch_result.insert(event_id, Ok(NotificationStatus::Event(event)));
-                    }
-                }
-                _ => {
-                    batch_result.insert(event_id, Ok(status));
-                }
+            if self.client.is_user_ignored(notification_item.event.sender()).await {
+                batch_result.insert(event_id, Ok(NotificationStatus::EventFilteredOut));
+            } else {
+                batch_result
+                    .insert(event_id, Ok(NotificationStatus::Event(Box::new(notification_item))));
```

Comment:
I suspect the modification might be incorrect.

The previous version of the code was:

```rust
match status {
    NotificationStatus::Event(event) => { … }
    _ => {
        batch_result.insert(event_id, Ok(status));
    }
}
```

The new version is:

```rust
if … { … }
else {
    batch_result.insert(event_id, Ok(NotificationStatus::Event(…));
}
```

It means that the semantics of the `else` case (for both the `match` in the previous version, or the `else` in the new version) aren't equivalent. In the former, we are inserting a status that is **not** `NotificationStatus::Event`, whilst in the other, we are always inserting a status of type `NotificationStatus::Event`.

Did I miss something? I'm not familiar with this code base.

Reply:
If you read the previous code:

```rust
let status =
                match NotificationItem::new(&room, raw_event, push_actions.as_deref(), Vec::new())
                    .await
                    .map(|event| NotificationStatus::Event(Box::new(event))) {
 Ok(status) => status,
 Err(err) => /* record the error and continue */
};

match status {
  NotificationStatus::Event(event) => do_some(),
  _ => {
   do_something_else();
  }
}
```

the `do_something_else()` arm is actually unreachable, because `status`, if we've reached that branch, is always a `NotificationStatus::Event`, because of the `.map()` call above.

The code was confusing, since it included bits that couldn't be reached. The new version avoids that confusion by removing this unreachable code path.

Reply:
Oh okay, I see. Thanks for the explanations!